tests/functional/test_article.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+
+from pages.article import ArticlePage
+
+
+# page title
+@pytest.mark.smoke
+@pytest.mark.nondestructive
+def test_title_in_title(base_url, selenium):
+    page = ArticlePage(selenium, base_url).open()
+    assert page.title_in_title(selenium)
+
+
+@pytest.mark.smoke
+@pytest.mark.nondestructive
+def test_title_text_has_length(base_url, selenium):
+    page = ArticlePage(selenium, base_url).open()
+    assert page.title_text_has_length(selenium)
+
+
+# article
+@pytest.mark.smoke
+@pytest.mark.nondestructive
+def test_article_is_displayed(base_url, selenium):
+    page = ArticlePage(selenium, base_url).open()
+    assert page.article_is_displayed
+
+
+# page buttons
+@pytest.mark.smoke
+@pytest.mark.nondestructive
+def test_language_menu_is_displayed(base_url, selenium):
+    page = ArticlePage(selenium, base_url).open()
+    assert page.language_menu_is_displayed
+
+
+@pytest.mark.smoke
+@pytest.mark.nondestructive
+def test_edit_button_is_displayed(base_url, selenium):
+    page = ArticlePage(selenium, base_url).open()
+    assert page.edit_button_is_displayed
+
+
+@pytest.mark.smoke
+@pytest.mark.nondestructive
+def test_advanced_menu_is_displayed(base_url, selenium):
+    page = ArticlePage(selenium, base_url).open()
+    assert page.advanced_menu_is_displayed
+
+
+"""
+The tests below are duplicates of ones in test_home, how can I avoid this?
```

Comment:
Some ideas.

Some duplication in tests is OK.  @jgmize says it is a good thing.  I try to remove as much as possible, but tests tend to evolve in interesting ways, and code that is identical today becomes different as new things are tested tomorrow.  Knowing when to allow duplication and when to remove it only comes with experience, and you still get it wrong half the time.

The first refactor would be to add the page class as a parameter:

```
@pytest.mark.smoke
@pytest.mark.nondestructive
def test_header_is_displayed(base_url, selenium, page_class=HomePage):
    page = page_class(selenium, base_url).open()
    assert page.Header.is_displayed

@pytest.mark.smoke
@pytest.mark.nondestructive
def test_article_header_is_displayed(base_url, selenium):
    test_header_is_displayed(base_url, selenium, page_class = ArticlePage
    page = page_class(selenium, base_url).open()
    assert page.Header.is_displayed
```

Maybe a whole line is saved!  In ``unittest``, a common pattern would be to collect related tests into a test class. Following a [blog post](http://theabele.com/dev/abstract-test-case-pytest):
```
class PageContract(object):

    @pytest.fixture
    def page_class(self):
        raise NotImplementedError('Need to override page_class')

    @pytest.mark.smoke
    @pytest.mark.nondestructive
    def test_header_is_displayed(self, base_url, selenium, page_class):
        page = page_class(selenium, base_url).open()
        assert page.Header.is_displayed

class TestArticlePageContract(PageContract):
    @pytest.fixture
    def page_class(self):
        return ArticlePage

class TestHomePageContract(PageContract):
    @pytest.fixture
    def page_class(self):
        return HomePage
```

This can potentially save more lines, but would involve rewriting test functions as classes and may confuse things.  There's also not a lot of +1 for that blog post, which may mean it isn't really the py.test philosophy.

Something closer to "good py.test code" may be move these page tests to a ``test_pages.py`` file, with something like:

```
@pytest.fixture(params=[ArticlePage, HomePage])
def page_class(request):
    return request.param

@pytest.mark.smoke
@pytest.mark.nondestructive
def test_header_is_displayed(base_url, selenium, page_class):
    page = page_class(selenium, base_url).open()
    assert page.Header.is_displayed
```

I think that means ``test_header_is_displayed`` will be called once with ``page_class=ArticlePage``, and once with ``page_class=HomePage``. I am a py.test newbie though, so I'm just going by the [fixture docs](http://doc.pytest.org/en/latest/fixture.html).

I'd be tempted to pursue one of these strategies, but that may just be to procrastinate on actually writing tests.  The wiser thing may be to accept the duplication for now and revisit once we have some months of experience.